backend/src/services/journal/journal.permissions.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+class JournalPermissions {
+  constructor(journalRepo, authRepo, coursePermissions) {
+    this.journalRepo = journalRepo;
+    this.authRepo = authRepo;
+    this.coursePermissions = coursePermissions;
```

Comment:
[Suggestion]
Instead of coupling with the course permissions API, we can take in the enrollment object and simply check the role through that.